### ISSUE TITLE
Setup ArangoDB Collections - Files, Segments, Parallels

### ISSUE THREAD

**Issue by ayya-vimala** [ayya-vimala/pali-networks#107]
New JSON files need to be loaded into ArangoDB (at first small test-batch for setting things up).
Collections in ArangoDB need to be set up according to earlier specs as attached.
[https://app.zenhub.com/files/181276680/0880dfe6-b660-4f3a-afa2-0be0f108271f/download](https://app.zenhub.com/files/181276680/0880dfe6-b660-4f3a-afa2-0be0f108271f/download)

As discussed with a difference to the parallels collection, which should be added to the new json files. The parallels collection does not need the parallel-segment text imported. It might be temporarily still in those files to show Orna.

Menu data is in src/suttalists/chinese-menu.js and tibetan-menu.js. The rest is not available yet.

**Comment by ayya-vimala** [comment]
Loading of the Menu data should already be possible for Tibetan and Chinese.
Right now the two menus are not consistent witch each other so I suggest just to import:
`searchString`,`textname`,`filename` and `category`. The Chinese also has fullname, author and dynasty but that info is also in the searchString and does not seem to be used anywhere else. @sebastian-nehrdich - can you comment on that?

**Comment by ayya-vimala** [comment]
Sample tibetan json files should be ready for loading in http://buddhist-db.de/vimala/new_suttas/tib

**Comment by sebastian-nehrdich** [comment]
Regarding the additional fields in the Chinese: I suggest to disregard them for the time being. Of course it would be excellent to have the user later filter for dynasty, author etc. as separate options, but I don't think we can pull this feature of so quickly, there are tons of issues waiting before we can get to that. I suggest to keep it in our reference jsons but not make use of it on the website for the time being if that is fine with you.

**Comment by TrebuhD** [comment]
Thank you @ayya-vimala for creating the tasks and giving them such great descriptions. I'm going to copy the collection specifications here from the file so that we can discuss the implementation:

1. MENU-COLLECTIONS
```
collection: “tengyur”
categories: [T01,T02,T03,T04,T05,T06, etc.]
language: “tib”
```
Where can I get this data? I have access to the language and can extract the `Txx` codes from the filenames, but I don't know how to find the collection name.

2. MENU-CATEGORIES 
```
category: T06
language: “tib”
categoryname: "Vijñaptimātra (Mind Only)"
files: ["T06TD4020E","T06TD4021E","T06TD40220E",etc.]
```

Same question here about the `categoryname`.

3. FILES
```
filename: “T06T4021E”
language: “tib”
searchString; “T06T4021E with some name if available”
textname: “D4021”
category: “T06”
segmentnrs: [0_0,2_0,3_1,4_1,5_1,6_0,7_0,8_0,etc.]
```

- What is `searchString` here?
- what is `textname` and how to find it?
- I'm assuming `category` is the first 3 characters in the filename?
- In the files, `segmentnr`s look like this: `K07kl00023e:8`. Why is this format different?

4. SEGMENTS
```
segmentnr: “T06T4021E:3_1”
language: “tib”
segment: "ma dhy'anta bi bham ga k'a ri k'a /"
```

I don't have any questions here :). This collection is already there, but split into three collections - by language. Maybe indeed it's a good idea to combine them into one.

5. PARALLELS
```
parsegnr: ["T06TD4027E:13_0","T06TD4027E:14_0","T06TD4027E:15_0"]
par_offset_beg: 0
par_offset_end: 7
root_segnr: ["T06TD4021E:8_0","T06TD4021E:9_0","T06TD4021E:10_0"]
root_offset_beg: 0
root_offset_end: 7
probability: 57
parlength: 6
co-occ: 2
```
Also no questions here, looks pretty straightforward. The parallel should have a unique ID.

I'm hoping to have this task done by the end of the weekend.

**Comment by sebastian-nehrdich** [comment]
Regarding (1) and (2), I can provide you with that data. Point is that we are still adjusting some things regarding this (I got some feedback from Orna that I need to integrate first). So once that is done, I wil upload this on github.
Also @ayya-vimala Regarding 5, it just came back to my mind that we might indeed better  record root_lang and par_lang because they can be different (I think we discussed that before). This becomes important on the frontend level because the languages need different steps of postprocessing. 
Also I want to add two more fields: root_string and par_string. These two strings will be just representing the parallel itself, but with some markup information - especially I want to mark words which are the same and words which are different, so we can display them on the frontend with some kind of color coding. This really becomes important for Tibetan, where all the syllables look pretty much the same. 
I think I can update the JSONS today with the above mentioned changes, if nobody has any objection to them.

**Comment by TrebuhD** [comment]
Thanks. We can either keep this data in the backend repository or on the server. If the files are reasonably small, I think the repository is a safer form of storage.

Just an idea, maybe the language could be deduced from the ID of the parallel / segment?

About the markup, just out of curiosity, are you planning to use HTML & CSS classes or some other format? 

One potential issue with that is, using `innerHTML` to inject the markup opens us to XSS attacks. We will be safe as long as we control the data though. I'll create a ticket to increase security of the database.

**Comment by ayya-vimala** [comment]
The language can be deduced from the ID indeed. There is a function:
```
export function checkLanguage(itemID) {
  let result = 'skt';
  if (
    itemID.includes('TD') ||
    itemID.includes('acip') ||
    itemID.match(/kl[0-9]/)
  ) {
    result = 'tib';
  } else if (itemID.includes('_T') || itemID.includes('_X')) {
    result = 'chn';
  }
  return result;
}
```

**Comment by sebastian-nehrdich** [comment]
Yes, that's enough I think. We just need to make sure this also works with
the Pali ids but I guess that's pretty easy.

**Comment by ayya-vimala** [comment]
If all sanskrit ID start with their respective collections T,K, GV,GP,GR or GS that should be easy.
The pali id are basically also the collection name with a number and are not in capitals.

**Comment by TrebuhD** [comment]
note: Moved to backlog - currently blocked by files not being accessible under `http://buddhist-db.de/suttas/tib/`.

**Comment by TrebuhD** [comment]
We agreed to move loading the first two collections (`menu-parallels` and `menu-categories`) into a separate ticket.

The PR for this ticket is here: https://github.com/ayya-vimala/buddhanexus/pull/1
